### ISSUE TITLE
ResolveAllReferences visitor misses some missing classes

### ISSUE THREAD

**Issue by dmivankov** [spotbugs/spotbugs#1606]
with `-visitors ResolveAllReferences`
https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ResolveAllReferences.java#L183

In following class
https://github.com/googleapis/java-spanner/blob/6e5d54fc71fd939874efc1596105a07d76a7a1bb/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java#L53
reference to `ErrorInfo` class is not inspected / not reported as missing class in case `ErrorInfo` is missing from dependencies.

With default set of visitors the class reference is inspected and reported if it's missing

**Comment by dmivankov** [comment]
following patch to ResolveAllReferences visitor fixes it
```diff
@@ -149,7 +149,7 @@ public class ResolveAllReferences extends PreorderVisitor implements Detector {
             }
             if (co instanceof ConstantClass) {
                 String ref = getClassName(obj, i);
-                if ((ref.startsWith("java") || ref.startsWith("org.w3c.dom")) && !defined.contains(ref)) {
+                if ((!ref.startsWith("java") || !ref.startsWith("org.w3c.dom")) && !defined.contains(ref) && !obj.getConstantPool().getConstantString(i, Const.CONSTANT_Class).startsWith("[")) {
                     bugReporter.reportBug(new BugInstance(this, "VR_UNRESOLVABLE_REFERENCE", NORMAL_PRIORITY).addClass(obj)
                             .addString(ref));
                 }

```
- looks like condition for "java" and "org.w3c.dom" was inverted - those refs were meant to be ignored?
- "[" ignore is a hack for primitive array references which also makes unnecessary ignore of non-primitive arrays https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.2
